WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift

```diff
@@ -194,13 +195,51 @@ extension ProductInventorySettingsViewController {
             self?.navigationController?.popViewController(animated: true)
         })
     }
+
+    private func enableDoneButton(_ enabled: Bool) {
+        navigationItem.rightBarButtonItem?.isEnabled = enabled
+    }
 }
 
 // MARK: - Input changes handling
 //
 private extension ProductInventorySettingsViewController {
     func handleSKUChange(_ sku: String?) {
         self.sku = sku
+
+        // If the sku is identical to the old one, is always valid.
+        guard sku != product.sku else {
+            skuIsValid = true
+            hideError()
+            throttler.cancel()
+            getSkuCell()?.textFieldBecomeFirstResponder()
+            enableDoneButton(true)
+            return
+        }
+
+        // Throttled API call
+        let siteID = self.siteID
+        throttler.throttle {
+            DispatchQueue.main.async {
+                let action = ProductAction.validateProductSKU(sku, siteID: siteID) { [weak self] isValid in
+                    guard let self = self else {
+                        return
+                    }
+                    self.skuIsValid = isValid
+
+                    guard isValid else {
+                        self.displayError(error: .duplicatedSKU)
+                        self.getSkuCell()?.textFieldBecomeFirstResponder()
+                        self.enableDoneButton(false)
+                        return
+                    }
+                    self.hideError()
+                    self.enableDoneButton(false)
```

Comment:
I think we might want to enable the Done button when the SKU is valid?

Reply:
OMG, I think I do a Ctrl+z before committing 💯  thanks for noticing it.